x/deployment/types/events_test.go

```diff
@@ -0,0 +1,187 @@
+package types
+
+import (
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pkg/errors"
+
+	"github.com/ovrclk/akash/sdkutil"
+)
+
+var (
+	keyAcc, _ = sdk.AccAddressFromBech32("akash1qtqpdszzakz7ugkey7ka2cmss95z26ygar2mgr")
+	//keyParams = sdk.NewKVStoreKey(params.StoreKey)
+
+	errWildcard = errors.New("wildcard string error can't be matched...")
+)
+
+type testEventParsing struct {
+	msg    sdkutil.Event
+	expErr error
+}
+
+func (tep testEventParsing) testMessageType() func(t *testing.T) {
+	_, err := ParseEvent(tep.msg)
+	return func(t *testing.T) {
```

Comment:
I'm guessing that you're returning a closure so that `t.Run()` executes in parallel... but `t.Parallel()` is not called and `ParseEvent()` wouldn't be run in it anyways.

I think just putting all of this in the `TestEventParsing` function instead of as a method would make this a lot easier to understand.

Or if you really want to use a method, put the closure in `TestEventParsing` instead of here.

Reply:
This pattern is purely due to the linter being tyranical about what state can be referenced in the main `t.Run(...)` function.  Nothing to do with t.Parallel().

Linter is harsh...